### DISCUSSION TITLE
function to switch operation mode on Defrost

### DISCUSSION THREAD

**Discussion by sabom2d** [edterbak/HeishaMoNR#146] [Q&A] [answered]
Hi,

I realized, that on Defrost I have Pressure surges in the pipe for Floor heating, I think because the HP raises the flowrate which I don't have any possibility to reduce as it seems.
I also realized, when I then switch oprating mode to DHW, I don't have any pressure surges anymore.
So my idea is to add an MQTT string in Node-red to switch operation mode to DHW when Defrost starts and to switch back to Heat, when defrost finishes. I'm already trying out myself how to set it up, but I'm a complete noob at node-red, so I would like to ask for support how to establish that function.

regards

ede

**Comment by edterbak** [comment]
Hi
In the settings page, the max flow rate, does it have any effect on the flow during defrost?

**Reply by sabom2d** [reply]
OK , my max flow rate value is empty, maybe because I'm still on 23.05.  Maybe that could also have an effect. Will check that on the next defrost cycle.
But I have a working solution now, but dont know, if it could have any side effect.
In Principle:
Top 4 as MQTT input, a function to switch 0/1 to 3/4 and Top9 as MQTT output. It's working as expected so far.
![AddonHPRWo](https://github.com/edterbak/NodeRed_Heishamon_control/assets/107671129/0d265520-8265-4b48-807c-4ca0806d709c)


Question now is, can I run this simple code or is there any restriction I need to include in the code?

regards

ede

**Comment by edterbak** [comment]
Second question, have you tried it out already during a defrost cycle to change the operating mode in the controller? Does it work?

**Reply by sabom2d** [reply]
Hi ed, yes changing the OP mode did change the behavior and pressure surge was gone directly, tested it last night on several Defrost cycles.

**Reply by sabom2d** [reply]
strangely, when I use FORCE DEFROST MODE now for Testing my script, it changes the OP Mode to DHW, but it seems also DFEROST is stopped, yesterday when the HP defrosted automatically, switching the OP Mode worked and HP was still defrosting.

**Comment by geduxas** [comment]
I think it would be best to predict defrost, and before defrost switch heating..

**Reply by sabom2d** [reply]
Yes, that would surely be a better solution, but how do I do this.
I tested last night again, strangely my script does not switch OP, but manually I can switch it and DEFROST keeps on running. 
Is there an additional check that needs to be included then just the TOP 26 state?

**Reply by geduxas** [reply]
Here is scenario for deicing from service manual.. 

![Screenshot_2023-11-26-12-55-21-69_e2d5b3f32b79de1d45acd1fad96fbb0f](https://github.com/edterbak/NodeRed_Heishamon_control/assets/18700450/105843b1-599d-414a-a5b0-7501916d74bc)

**Reply by sabom2d** [reply, marked as answer]
Hi Ed,

I'm not sure why, but after 2 days now my script finnally started to work as expected, as soon aos DEFROST is triggert, OP-Mode switches to DHW and on DEFROST Finish, OP-Mode switches back to HEAT. I don't see why it did not work directly as I didn't change it the last days. But I see the root cause is my old existing pump in the Heating system, it does not hande the high volume of the HP-internal Pump when DEFROSTING. So I will try to find a replacement pipe for that pump and see, if the internal HP pump is strong enough.

Thanks.

regards

ede
